drivers/serial/uart_native_pty.c

```diff
@@ -14,12 +14,17 @@
 #endif
 
 #include <stdbool.h>
-#include <zephyr/drivers/uart.h>
+#include <string.h>
+#include <errno.h>
+#include <unistd.h>
```

Comment:
Drivers like this for native_sim are split in 2 parts.
One part (this file) is built with the embedded code.
The other part (the _bottom) is build in the host/Linux context.
The embedded code is by default in some cases built with the host/Linux libC, which means this code can work in those cases.
But the embedded code is also very often built with an embedded libC (set CONFIG_PICOLIBC=y for ex.). When one does that, this file will not see the host headers, and calls to the host libC will be instead hitting the embedded libC. 
That is why we have the bottom part, built in the "host context".
We do all the interaction with the host OS/Linux from the "bottom" part of the driver. And we have this top side call into the bottom.